src/github.com/matrix-org/dendrite/syncserver/storage/output_room_events_table.go

```diff
@@ -244,7 +267,14 @@ func rowsToEvents(rows *sql.Rows) ([]gomatrixserverlib.Event, error) {
 		if err != nil {
 			return nil, err
 		}
-		result = append(result, ev)
+		if order == rowOrderForward {
+			result = append(result, ev)
+		} else if order == rowOrderReverse {
+			result = append([]gomatrixserverlib.Event{ev}, result...)
```

Comment:
Either way, it's gone now. https://github.com/matrix-org/dendrite/pull/72/commits/f62751ef10964d7b69afa6150917a2816f27f51f